inlinino/instruments/hydroscat.py

```diff
@@ -28,15 +29,14 @@ class HydroScat(Instrument):
                            'output_cal_header',
                            'variable_names', 'variable_units', 'variable_precision']
 
-    def __init__(self, uuid, signal, *args, **kwargs):
+    def __init__(self, uuid, cfg, *args, **kwargs):
+        super().__init__(uuid, cfg, *args, setup=False, **kwargs)
         # Instrument state machine
+        self.hydroscat: aquasense.hydroscat.HydroScat = None
```

Comment:
I'm curious about this line. Not familiar with this syntax. The denotation appears to be something like:
~~~
aquasense.hydroscat.HydroScat = None if self.hydroscat
~~~
but I suspect that's not it.

Should this assignment to `None` be in the `try...import hydroscat...except` block?

Reply:
This syntax is used for type hints in python (c.f. [typing](https://docs.python.org/3/library/typing.html)), it's not used by the Python runtime but can be used by your IDE to make suggestions and check variable types for you. The syntax is as follow:

    variable_name: variable_type = variable_value

In this case, it's equivalent to `self.hydroscat = None`. Omitting `:aquasense.hydroscat.HydroScat` won't change how the Python runtime execute the code.

I did set `self.hydroscat` in the __init__ function, as a personal preference and for consistency with the rest of Inlinino's code base (all class variables are defined in `__init__`), the code would still run fine without setting up the variable. The value is changed during initialization as `self.setup()` is called during `__init__`.

Reply:
@doizuc, I was obviously not thinking at all clearly when I looked at that line of code.

In fact I use and advocate for type hints in Python all the time in my code (e.g. see https://bitbucket.csiro.au/projects/AQRS/repos/aquasense/browse/src/aquasense/hydroscat/extract_raw.py).

I can only assume that my confusion arose from its use in an instance variable since I have probably only used it in function/method signatures so far.

Also, I completely agree with you about initialising all instance variables. I am sometimes less strict about this in Python than in other languages like C++, C# or Java, for no very good reason, so thanks for picking me up on that.